mantidproject/mantid
# Indirect Reduction Sum files bug with wrong masked detectors

**Description of work.**
When doing an individual run, it finds the masked detectors of the first detector workspace and so it doesn’t know about the extra masked detector in the 4th detector workspace. This means there is a masked detector missing!
When doing `Sum Files`, it will have summed all the runs and so it finds all of the masked detectors (including the extra masked detector in detector workspace 4). 

Having this missing masked detector for the individual runs meant there is a discrepancy at around 8cm-1 on the x axis of the output plot.
In the second example, the masked detectors for the summed run was incorrect. The masked detectors of the summed run should be the summed masked detectors of the individual runs it is made up of!

**To test:**
1.`Interfaces`->`Indirect`->`Data Reduction`->`Energy Transfer`
2. Enter run numbers `22797-22799`
3. Tick `Sum Files` and Run
4. Untick `Sum Files` and Run
5. Select all the workspaces in the ADS and plot the second spectrum
6. Change the x axis range to be -10 to 20
7. Change the y axis range to be 0 to 20

The output plot should look like:
![image](https://user-images.githubusercontent.com/40830825/53969850-6eb28500-40f1-11e9-9e4f-a78e80e53119.png)

Before this fix the plot looked like:
![plotunfixed](https://user-images.githubusercontent.com/40830825/53969869-783bed00-40f1-11e9-9b62-be7d49c96857.jpg)

1.`Interfaces`->`Indirect`->`Data Reduction`->`Energy Transfer`
2. Enter run numbers `22841-22843`
3. Tick `Sum Files` and Run
4. Untick `Sum Files` and Run
5. Select all the **new** workspaces in the ADS and plot the second spectrum
6. Change the x axis range to be 0 to 4000
7. Change the y axis range to be 0 to 1

The output plot should look like:
![image](https://user-images.githubusercontent.com/40830825/54018023-92241100-417f-11e9-9407-44eb87cbfca7.png)

Before this fix the plot looked like:
![plot3unfixed](https://user-images.githubusercontent.com/40830825/54025949-13859e80-4194-11e9-9905-43913d953ccb.jpg)

Fixes #25061 and #25072

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
